src/library.py

```diff
@@ -910,82 +912,259 @@ def verify(self):
+                df_s_AHU_diff = day[day["s_AHU"].diff() != 0.0]["s_AHU"]
+                if len(df_s_AHU_diff) > 1:
+                    # unocc to occ
+                    df_occ_diff = day[day["occ"].diff() != 0.0]["occ"]
+
+                    if (
+                        len(df_occ_diff) > 1
+                    ):  # Since first value of .diff() is nan, this means when values don't change
+                        t_unocc_to_occ = df_occ_diff.index.tolist()[1]
                     else:
-                        result_repo.append(
-                            0
-                        )  # fail, zone temperature setpoint was not reset correctly
-                else:  # room is rented out
-                    T_z_hea_occ_set = day.query("O_sch > 0.0")["T_z_hea_set"].max()
-                    T_z_coo_occ_set = day.query("O_sch > 0.0")["T_z_coo_set"].min()
+                        t_unocc_to_occ = day.index[0]
+
+                    t_s_AHU_off_to_on = df_s_AHU_diff.index.tolist()[1]
 
                     if (
-                        day["T_z_hea_set"] < T_z_hea_occ_set - 2.22 + tol_temp
-                    ).all() or (
-                        day["T_z_coo_set"] > T_z_coo_occ_set + 2.22 - tol_temp
-                    ).all():
-                        result_repo.append(
-                            1
-                        )  # pass, confirm the HVAC setpoint control resets when guest room reset when occupants leave the room
+                        pd.Timedelta(t_s_AHU_off_to_on - t_unocc_to_occ).seconds / 3600
+                        > 12
+                    ):
+                        t_length.append(
+                            pd.Timedelta(t_unocc_to_occ - t_s_AHU_off_to_on).seconds
+                        )
                     else:
-                        result_repo.append(
-                            0
-                        )  # fail, reset does not meet the standard or no reset was observed.
-                year_info = day.index.year[0]
+                        t_length.append(
+                            pd.Timedelta(t_s_AHU_off_to_on - t_unocc_to_occ).seconds
+                        )
+
+                    T_oa_dry_repo.append(day.loc[t_s_AHU_off_to_on]["T_oa_dry"])
+                    T_z_measure = day.loc[t_unocc_to_occ]["T_z_measure"]
+                    T_diff_heating.append(
+                        T_z_measure - day.loc[t_unocc_to_occ]["T_z_hea_set"]
+                    )
+                    T_diff_cooling.append(
+                        abs(T_z_measure - day.loc[t_unocc_to_occ]["T_z_coo_set"])
+                    )
+            year_info = day.index.year[0]
+
+        if len(self.df["s_AHU"].unique()) == 1 and self.df["s_AHU"].unique() in [0, 1]:
+            if self.df["s_AHU"].unique() == 0:
+                self.df["result"] = 0
+                self.msg = "NO optimum start"
+            else:
+                if (
+                    pearsonr(t_length, T_oa_dry_repo)[0] > 0.5
```

Comment:
That seems like a really low threshold. What did you get in practice?

Reply:
Thanks for pointing this out, Jeremy. 

First of all, this value is from Yan's pseudo code.  I believe 0.5 value should be tested with more datasets thoroughly because although this item passed because of (heating setpoint corr: 0.8, cooling setpoint corr: 0.7), the ```T_oa_dry``` corr was -0.5. The -0.5 value itself made sense considering the fact that the building's location is in Denver.

I think the pseudo-code makes sense the 0.5 value criteria should be discussed/tested more in the future.